.github/workflows/sycl-linux-build.yml

```diff
@@ -339,6 +342,25 @@ jobs:
         sycl_compiler: $GITHUB_WORKSPACE/toolchain/bin/clang++
         extra_lit_opts: --param sycl_build_targets="spir;nvidia;amd"
 
+    - name: Remove E2E tests before NewOffloadModel run
+      if: ${{ inputs.e2e_binaries_new_offload_model_artifact && !cancelled() && steps.build.conclusion == 'success' }}
+      run: rm -rf build-e2e
+
+    - name: Build E2E tests with NewOffloadModel
+      if: |
+        inputs.e2e_binaries_new_offload_model_artifact &&
+        !cancelled()
+        && steps.build.conclusion == 'success' &&
+        contains(github.event.pull_request.labels.*.name, 'new-offload-model')
+      uses: ./devops/actions/run-tests/e2e
+      with:
+        ref: ${{ inputs.ref || github.sha }}
+        testing_mode: build-only
+        target_devices: all
+        binaries_artifact: ${{ inputs.e2e_binaries_new_offload_model_artifact }}
+        sycl_compiler: $GITHUB_WORKSPACE/toolchain/bin/clang++
+        extra_lit_opts: --param sycl_build_targets="spir;nvidia;amd" --param enable_new_offload_model=True
```

Comment:
This is getting too verbose. Can we fold the removal into the `devops/actions/run-tests/e2e` and then use a matrix job to build all the flavors without that much copy-paste?

Reply:
done.